### DISCUSSION TITLE
2.30.0

### DISCUSSION THREAD

**Discussion by millsp** [prisma/prisma#8892] [Q&A]
Today, we are excited to share the `2.30.0` stable release 🎉

🌟 **Help us spread the word about Prisma by starring the repo or [tweeting](https://twitter.com/intent/tweet?text=Check%20out%20the%20latest%20@prisma%20release%20v2.30.0%20%F0%9F%9A%80%0D%0A%0D%0Ahttps://github.com/prisma/prisma/releases/tag/2.30.0) about the release.** 🌟 

### Full-Text Search for PostgreSQL is now in Preview 🚀

We're excited to announce that Prisma Client now has [preview support](https://www.prisma.io/docs/about/prisma/releases#preview) for Full-Text Search on PostgreSQL. 

You can give this a whirl in **2.30.0** by enabling the `fullTextSearch` preview flag:

```prisma
datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextSearch"]
}

model Post {
  id     Int    @id @default(autoincrement())
  title  String @unique
  body   String
  status Status
}

enum Status {
  Draft
  Published
}
```

After you regenerate your client, you'll see a new `search` field on your `String` fields that you can query on. Here are a few examples:

```ts
// returns all posts that contain the words cat *or* dog.
const result = await prisma.posts.findMany({
  where: {
    body: {
      search: 'cat | dog',
    },
  },
})

// All drafts that contain the words cat *and* dog.
const result = await prisma.posts.findMany({
  where: {
    status: "Draft",
    body: {
      search: 'cat & dog',
    },
  },
})
```

You can learn more about how the query format works in [our documentation](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search). We would love to know your feedback! If you have any comments or run into any problems we're available in this [in this Github issue](https://github.com/prisma/prisma/issues/8877). 

### Validation errors for referential action cycles on Microsoft SQL Server ℹ

Microsoft SQL Server has validation rules for your schema migrations that reject schema changes that introduce *[referential action cycles](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions/cyclic-referential-actions)*.
These scenarios tend to show up often for developers using the `referentialActions` preview feature, which will become the default. The database error you get is not really helpful, so to provide a better experience, Prisma now checks for referential cycle actions when it validates your schema file and shows you the exact location of the cycle in your schema.

To learn more, check out the [documentation](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions/cyclic-referential-actions).

### `prisma introspect` is being deprecated in favor of `prisma db pull` 👋🏻

The `prisma introspect` is an alias for `prisma db pull` so  they are the same command.  However,  `prisma db pull` is more intuitive since it _pulls_ the schema from the database into your local `schema.prisma` file. This naming also works as the counterpart of `prisma db push`.

Starting with this release, you will get a warning that encourages you to use `prisma db pull` instead of `prisma introspect`.

### Prisma Client

- [Revisit prisma.$queryRaw("...") vs. prisma.$queryRaw template literal](https://github.com/prisma/prisma/issues/7142)
- [Queries that include the count of a relation that has multiple foreign keys translates to incorrect SQL](https://github.com/prisma/prisma/issues/7299)
- [Order by Relations with multiple references](https://github.com/prisma/prisma/issues/7331)
- [Enabling Node-API libraries: Remove `nApi` preview feature flag and make Node-API libraries the new default, provide escape hatch to old behavior](https://github.com/prisma/prisma/issues/7561)
- [Prisma client crashes when counting m-n self-relation](https://github.com/prisma/prisma/issues/7807)
- [Prisma crashes when I use the same table for counting and `findUnique`](https://github.com/prisma/prisma/issues/7893)
- [Order By Relation with CursorId doesn't return result](https://github.com/prisma/prisma/issues/8036)
- [Shows incorrect count number by using selectRelationCount previewFeatures](https://github.com/prisma/prisma/issues/8050)
- [JSON number filtering bug](https://github.com/prisma/prisma/issues/8224)
- [MDBE: Break down the feature into concrete tasks](https://github.com/prisma/prisma/issues/8275)
- [FTS: Integrate into the Prisma Client](https://github.com/prisma/prisma/issues/8694)
- [Interactive transactions not working with `$queryRaw` and `$executeRaw` in 2.29.0](https://github.com/prisma/prisma/issues/8717)
- [Calling `.catch()` or `.finally()` on a prisma client model query does not fire the request in 2.29.0](https://github.com/prisma/prisma/issues/8718)

### Prisma Migrate

- [CLI: Add deprecation notice to prisma introspect (renamed db pull)](https://github.com/prisma/prisma/issues/8078)
- [Add Default Constraint Name to PSL for SqlServer](https://github.com/prisma/prisma/issues/8158)
- [The server failed to resume the transaction. Desc:3600000005.](https://github.com/prisma/prisma/issues/8394)
- [Add DML validation to reject `Bytes` `@id` columns with SQL Server](https://github.com/prisma/prisma/issues/8591)
- [Better error for cyclical referential actions](https://github.com/prisma/prisma/issues/8682)
- [Sqlite reserved constraint names cannot be recreated](https://github.com/prisma/prisma/issues/8759)

### Language tools (e.g. VS Code)

- [Autocomplete does not work for arguments after the first one in `@relation()`](https://github.com/prisma/language-tools/issues/814)
- [referentialActions: No auto complete for available actions](https://github.com/prisma/language-tools/issues/818)

### @prisma/engines npm package

- [Cannot find namespace 'debug' in 2.29.0](https://github.com/prisma/engines-wrapper/issues/128)

## Credits

Huge thanks to @saintmalik for helping!

## 📺 Join us for another "What's new in Prisma" livestream

Learn about the latest release and other news from the Prisma community by joining us for another ["What's new in Prisma"](https://www.youtube.com/watch?v=TUu4h0elhpw&ab_channel=Prisma) livestream.

The stream takes place [on Youtube](https://www.youtube.com/watch?v=TUu4h0elhpw&ab_channel=Prisma) on **Thursday, August 26** at **5pm Berlin | 8am San Francisco**.

<hr /><em>This discussion was created from the release <a href='https://github.com/prisma/prisma/releases/tag/2.30.0'>2.30.0</a>.</em>

**Comment by saintmalik** [comment]
🎉🚀🚀, Thanks for the mention, am really glad to help.

**Comment by gautvm** [comment]
Awesome release! This release was perfect timing for me, I was just about to implement searching in my app!

**Reply by leerobert** [reply]
I was literally just reading how to get an elasticsearch cluster up and rolling... this feature is gonna take prisma to the next level. 

This dev community is AMAZING.

**Comment by hienlh** [comment]
Awesome search feature! 
Thanks so much for working hard.

**Comment by PatElder** [comment]
Breaks Prisma Studio on Windows

**Reply by janpio** [reply]
Thanks, reported and tracked here: https://github.com/prisma/studio/issues/767

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
